### PR TITLE
feat: add permission mapper for media browser

### PR DESCRIPTION
**Pull request by gabrielfs7**
feat: add permission mapper for media browser

https://oat-sa.atlassian.net/browse/ADF-271

- [ ] https://github.com/oat-sa/extension-tao-item/pull/470
- [ ] https://github.com/oat-sa/extension-tao-mediamanager/pull/269

### CODE REVIEW

**Review by voknil — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/media/mapper/MediaBrowserPermissionsMapper.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        if ($this->hasWriteAccess($resourceUri)) {
+            $data[self::DATA_PERMISSIONS][] = PermissionCheckerInterface::PERMISSION_WRITE;
+        }
+
+        return $data;
+    }
+
+    private function getPermissionChecker(): PermissionCheckerInterface
+    {
+        if (!$this->permissionChecker) {
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/media/mapper/MediaBrowserPermissionsMapper.php:64]: yes

File: models/classes/media/mapper/MediaBrowserPermissionsMapper.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    private function hasReadAccess(string $uri): bool
+    {
+        return $this->enableAccessControl === true
+            ? $this->getPermissionChecker()->hasReadAccess($uri)
+            : true;
+    }
+
+    private function hasWriteAccess(string $uri): bool
+    {
+        return $this->enableAccessControl === true
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/media/mapper/MediaBrowserPermissionsMapper.php]: Ok, moved to a private method

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**